assets/js/pages/ActivityLogPage/ActivityLogPage.jsx

```diff
@@ -37,19 +42,47 @@ const filters = [
   },
 ];
 
+const defaultItemsPerPage = 20;
+const itemsPerPageOptions = [10, 20, 50, 75, 100];
+
+const changeItemsPerPage = (searchParams) => (items) => {
+  if (searchParams.has('after')) {
+    return {
+      first: items,
+      after: searchParams.get('after'),
+    };
+  }
+  if (searchParams.has('before')) {
+    return {
+      last: items,
+      before: searchParams.get('before'),
+    };
+  }
+  return { first: items };
+};
+
 function ActivityLogPage() {
-  const [searchParams, setSearchParams] = useSearchParams();
+  const [searchParams, setSearchParams] = useSearchParams(
+    resetPaginationToSearchParams(defaultItemsPerPage)()
+  );
   const [activityLog, setActivityLog] = useState([]);
+  const [activityLogMetadata, setActivityLogMetadata] = useState({});
+
   const [isLoading, setLoading] = useState(true);
   const [activityLogDetailModalOpen, setActivityLogDetailModalOpen] =
     useState(false);
 
+  const itemsPerPage = pipe(getItemsPerPageFromSearchParams, (n) =>
+    itemsPerPageOptions.includes(n) ? n : itemsPerPageOptions[0]
+  )(searchParams);
+
   const fetchActivityLog = () => {
     setLoading(true);
     const params = searchParamsToAPIParams(searchParams);
     getActivityLog(params)
-      .then((response) => {
-        setActivityLog(response.data?.data ?? []);
+      .then(({ data: { data = [], ...metadata } = {} }) => {
```

Comment:
**suggestion:** what about destructuring `pagination` or `paginationMetadata` to not create confusion with a log entry metadata? 
```suggestion
      .then(({ data: { data = [], ...pagination } = {} }) => {
```
Also related state and state setter `activityLogMetadata` and `setActivityLogMetadata`

Reply:
`metadata` is an object that contains all the sub-objects including the pagination. If we rename like this we will have `pagination.pagination`. Maybe we can rename it like

```js
      .then(({ data: { data = [], pagination } = {} }) => {
```

It's what you intended?

Reply:
otherwise, if we need/want to keep access to the whole rest of the response, we could actually go after what it is: `activityLogResponse` or the like.

Reply:
Good, I think it's more neat. Done in https://github.com/trento-project/web/pull/2930/commits/c8d3e60c918dc593fe8333de36ca3cb6b503ddc6